rust/signer/src/metadata.rs

```diff
@@ -0,0 +1,134 @@
+use hex;
+use codec::{Encode, Decode};
+use frame_metadata::{RuntimeMetadataV12, DecodeDifferent};
+use serde_json;
+use blake2_rfc::blake2b::blake2b;
+
+/// struct to decode the version metadata
+
+#[derive(Debug, Encode, Decode)]
+pub struct VersionDecoded {
+    pub specname: String,
+    implname: String,
+    auth_version: u32,
+    pub spec_version: u32,
+    impl_version: u32,
+    apis: Vec<(u8, u32)>,
+    trans_version: u32,
+}
+
+/// function takes full metadata in format '0x******', decodes
+/// using RuntimeMetadataV12, finds version in constants block
+/// within system module, and outputs version as a decodeable 
+/// vector; some checking done along the way
+
+pub fn get_meta_const (meta: &str) -> Result<Vec<u8>, &str> {
+
+    if !meta.starts_with("0x6d657461") {
+        return Err("No 'meta' starting sequence in metadata");
+    }
+    
+    let part1 = &meta[10..12];
+    let part1_vec = hex::decode(part1).unwrap();
+    let part1_decoded = u8::decode(&mut &part1_vec[..]).unwrap();
+    if part1_decoded != 12 {
```

Comment:
Does this mean when v13 comes out we no longer will be able to update the metadata?

Reply:
> Does this mean when v13 comes out we no longer will be able to update the metadata?

Actually, as far as I understand, losing compatibility on major API change is safety feature here, but confirming this statement wait until our secure interface audit next month.